pre_commit_hooks/destroyed_symlinks.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import sys
+from operator import methodcaller
+from subprocess import check_call
+from subprocess import check_output
+from typing import Optional
+from typing import Sequence
+
+ORDINARY_CHANGED_ENTRIES_MARKER = b'1'
+PERMS_LINK = b'120000'
+PERMS_NONEXIST = b'000000'
+
+
+def normalize_content(content: bytes) -> bytes:
+    return b'\n'.join(
+        filter(
+            None,
+            map(
+                methodcaller('strip'),
+                content.splitlines(),
+            ),
+        ),
+    )
+
+
+def find_destroyed_symlinks(autofix: bool) -> Sequence[bytes]:
+    destroyed_links = []
+    for line in check_output(
+        ['git', 'status', '--porcelain=v2', '-z'],
+    ).split(b'\0'):
+        splitted = line.split(b' ')
+        if splitted and splitted[0] == ORDINARY_CHANGED_ENTRIES_MARKER:
+            # variable names are taken from
+            # https://git-scm.com/docs/git-status#_changed_tracked_entries
+            _, XY, sub, mH, mI, mW, hH, hI, *path_splitted = splitted
+            path = b' '.join(path_splitted)
+            if all((
+                mH == PERMS_LINK,
+                mI != PERMS_LINK,
+                mI != PERMS_NONEXIST,
+            )):
+                found_destroyed_link = False
+                if hH == hI:
+                    # if old and new hashes are equal, it's not needed to check
+                    # anything more, we've found a destroyed symlink for sure
+                    found_destroyed_link = True
+                else:
+                    # if old and new hashes are *not* equal, it doesn't mean
+                    # that everything is OK - new file may be altered
+                    # by something like trailing-whitespace and/or
+                    # mixed-line-ending hooks so we need to go deeper
+                    index_size = int(
+                        check_output(['git', 'cat-file', '-s', hI]).strip(),
```

Comment:
there's a `cmd_output` helper in `pre_commit_hooks/util.py` that probably makes most of this easier to read

Reply:
```
    kwargs.setdefault('stderr', subprocess.PIPE)
```
This eats stderr output which is probably not the best idea.

Reply:
that's just the default, you can configure it otherwise.  and actually it's probably better as a pipe so the errors only display when the tool breaks